api/src/main/scala/hmda/api/http/LarHttpApi.scala

```diff
@@ -80,20 +79,27 @@ trait LarHttpApi extends LarProtocol with ValidationResultProtocol with HmdaCust
       }
     }
 
-  def validateRoute(lar: LoanApplicationRegister, checkType: String): scaladsl.server.Route = {
-    val larValidation = system.actorSelection("/user/larValidation")
+  def validateRoute(lar: LoanApplicationRegister, checkType: String, path: String) = {
+    val supervisor = system.actorSelection("/user/supervisor")
+    val fLarValidation = (supervisor ? FindActorByName(SingleLarValidation.name)).mapTo[ActorRef]
     val vContext = ValidationContext(None, None)
     val checkMessage = checkType match {
       case "syntactical" => CheckSyntactical(lar, vContext)
       case "validity" => CheckValidity(lar, vContext)
       case "quality" => CheckQuality(lar, vContext)
       case _ => CheckAll(lar, vContext)
     }
-    onComplete((larValidation ? checkMessage).mapTo[ValidationErrors]) {
-      case Success(xs) =>
-        complete(ToResponseMarshallable(aggregateErrors(xs)))
-      case Failure(e) =>
-        complete(HttpResponse(StatusCodes.InternalServerError))
+    val validationErrors = for {
+      larValidation <- fLarValidation
+      ve <- (larValidation ? checkMessage).mapTo[ValidationErrors]
+    } yield ve
+
+    onComplete(validationErrors) {
+      case Success(validationErrors) =>
```

Comment:
Good catch! I renamed the first `validationErrors` to `fValidationErrors`, to signify that it was a future.